Controller.cpp

```diff
@@ -523,6 +591,28 @@ namespace Procon {
 		return sendCommand(0x10, buf);
 	}
 
+	Controller::exchangeArray Controller::getSpiData(uint32_t offset, const uint8_t readLen)
+	{
+		std::array<uint8_t, 0x100> buf;
+		while (true)
+		{
+			buf.fill(0);
+			*(uint32_t*)(buf.data()) = offset;
+			*(uint8_t*)(buf.data() + 4) = readLen;
+			auto res = sendSubcommand(0x01, 0x10, buf);
```

Comment:
Does the procon expect 0x100 bytes here? If not you should use a smaller `array` since `buf` is never wrote to except by the three lines above this.

Reply:
I think we can absolutely use a smaller value here. I think I saw it in another implementation and did not thought about it further.

Reply:
49 (x31) bytes is the **max** expected buffer received for x01 cmd. Anything more gets trimmed by procon.
49 (x31) bytes is the max amount for the x01 subcmd reply from procon.

So 49 bytes are enough. 

----
FYI,
The 37bytes for subcmd reply **data**, I wrote in Nintendo_Switch_Reverse_Engineering, includes the 2 bytes subcmd ack.
It's a left over from the old layout. The correct there is 35 bytes.